crates/telio-nurse/src/qos.rs

```diff
@@ -181,9 +199,16 @@ impl Runtime for Analytics {
                 })
             },
 
-            _ = self.rtt_interval.tick() => {
+            _ = self.rtt_interval.tick(), if self.ping_channel_tx.upgrade().is_none() => {
```

Comment:
Can you clarify what kind of situation is this guard preventing?

Reply:
I didn't think much about it honestly and followed the same pattern, in fact I'm wondering if one of these select branches could be interrupted since it we're implementing wait() fn and not wait_with_update()